src/mahoji/commands/leagues.ts

```diff
@@ -0,0 +1,103 @@
+import { Time } from 'e';
+import { ApplicationCommandOptionType, CommandRunOptions } from 'mahoji';
+
+import { production } from '../../config';
+import {
+	allLeagueTasks,
+	generateLeaguesTasksTextFile,
+	leaguesCheckUser,
+	leaguesClaimCommand,
+	leagueTasks
+} from '../../lib/leagues/leagues';
+import { formatDuration } from '../../lib/util';
+import { Cooldowns } from '../lib/Cooldowns';
+import { OSBMahojiCommand } from '../lib/util';
+
+export const leaguesCommand: OSBMahojiCommand = {
+	name: 'leagues',
+	description: 'Manage your Leagues progress.',
+	options: [
+		{
+			type: ApplicationCommandOptionType.Subcommand,
+			name: 'check',
+			description: 'Check your current progress.'
+		},
+		{
+			type: ApplicationCommandOptionType.Subcommand,
+			name: 'view_task',
+			description: 'View/search a specific task.',
+			options: [
+				{
+					type: ApplicationCommandOptionType.String,
+					name: 'task',
+					description: 'Search for the task name.',
+					required: true,
+					autocomplete: async (value: string) => {
+						return allLeagueTasks
+							.filter(i => (!value ? true : i.name.toLowerCase().includes(value.toLowerCase())))
+							.map(i => ({ name: i.name, value: i.id.toString() }));
+					}
+				}
+			]
+		},
+		{
+			type: ApplicationCommandOptionType.Subcommand,
+			name: 'claim',
+			description: 'Claim the points from your completed tasks.'
+		},
+		{
+			type: ApplicationCommandOptionType.Subcommand,
+			name: 'view_all_tasks',
+			description: 'View all tasks.',
+			options: [
+				{
+					type: ApplicationCommandOptionType.Boolean,
+					name: 'exclude_finished',
+					description: 'Exclude tasks you have finished?'
+				}
+			]
+		}
+	],
+	run: async ({
+		options,
+		userID
+	}: CommandRunOptions<{
+		check?: {};
+		view_task?: { task: string };
+		claim?: {};
+		view_all_tasks?: { exclude_finished?: boolean };
+	}>) => {
+		const cooldown = Cooldowns.get(userID.toString(), 'leagues', Time.Second * 30);
+		if (cooldown && production) {
```

Comment:
Maybe move the cooldown + checkUser a bit lower if possible. I'm not sure if the `view all tasks` / `view task` options require this data, but if it does, then disregard.